pkg/apis/cluster.cattle.io/v3/types.go

```diff
@@ -14,11 +14,11 @@ type ClusterUserAttribute struct {
 	metav1.TypeMeta   `json:",inline"`
 	metav1.ObjectMeta `json:"metadata,omitempty"`
 
-	Groups       []string            `json:"groups,omitempty"`
-	LastRefresh  string              `json:"lastRefresh,omitempty"`
-	NeedsRefresh bool                `json:"needsRefresh"`
-	Enabled      bool                `json:"enabled"`
-	Extra        map[string][]string `json:"extra,omitempty"`
+	Groups          []string                       `json:"groups,omitempty"`
+	LastRefresh     string                         `json:"lastRefresh,omitempty"`
+	NeedsRefresh    bool                           `json:"needsRefresh"`
+	Enabled         bool                           `json:"enabled"`
+	ExtraByProvider map[string]map[string][]string `json:"extraByProvider,omitempty"`
```

Comment:
I am not 100% sure where all ClusterUserAttribute is used, but we decided to deprecate `Extra` field. Are there any use cases we should be concerned/handling with this?

Reply:
ClusterUserAttribute is used by the ACE endpoint and its webhook. It had the field Extra, but was not set and used until now by ACE codebase https://github.com/rancher/kube-api-auth/blob/ca50db33ff69ec47636ae21f714ab741750d58bf/pkg/service/handlers/v1-kube-api-authn.go

The change in this PR will help us fix this Audit for ACE path as well in future https://github.com/rancher/rancher/issues/33330


Rancher has the [controllers](https://github.com/rancher/rancher/blob/release/v2.6/pkg/controllers/managementuser/clusterauthtoken/user_attribute.go#L26) to keep the data in ClusterUserAttribute in sync with the v3.UserAttribute CR which holds information about the user principals. In this PR I have added the `ExtraByProvider` newly to the v3.UserAttribute - it was not used before.